ext/websocket/02_websocketstream.js

```diff
@@ -440,7 +465,58 @@ class WebSocketStream {
     );
   }
 }
-
 const WebSocketStreamPrototype = WebSocketStream.prototype;
 
-export { WebSocketStream };
+class WebSocketError extends DOMException {
+  #closeCode;
+  #reason;
+
+  constructor(message = "", init = {}) {
+    super(message, "WebSocketError");
+    this[webidl.brand] = webidl.brand;
+
+    init = webidl.converters["WebSocketCloseInfo"](
+      init,
+      "Failed to construct 'WebSocketError'",
+      "Argument 2",
+    );
+
+    if (init.reason && init.closeCode) {
+      init.closeCode = 1000;
+    }
+
+    this.#closeCode = init.closeCode;
+    this.#reason = init.reason;
```

Comment:
We're missing https://whatpr.org/websockets/48.html#validate-close-code-and-reason here